baremetal/metal3machine_manager.go

```diff
@@ -576,8 +585,19 @@ func (m *MachineManager) exists(ctx context.Context) (bool, error) {
 // getHost gets the associated host by looking for an annotation on the machine
 // that contains a reference to the host. Returns nil if not found. Assumes the
 // host is in the same namespace as the machine.
-func (m *MachineManager) getHost(ctx context.Context) (*bmh.BareMetalHost, error) {
-	annotations := m.Metal3Machine.ObjectMeta.GetAnnotations()
+func (m *MachineManager) getHost(ctx context.Context) (*bmh.BareMetalHost, *patch.Helper, error) {
```

Comment:
a bit confused why do we need two functions with same name, if the idea is to get helper also, may be the naming could be changed accordingly?

Reply:
one of them takes the manager, the other is a helper, that can be used in other places. Having the same name is not a problem, and IMO it clearly shows the flow, you call the gethost of the manager, that under the hood calls the helper of the same name